lib/modules/platform/codecommit/iam-client.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  GetUserCommand,
+  GetUserCommandOutput,
+  IAMClient,
+} from '@aws-sdk/client-iam';
+import type { Credentials } from '@aws-sdk/types';
+import { regEx } from '../../../util/regex';
+
+let iam: IAMClient;
+
+export function initIamClient(
+  region: string,
+  credentials: Credentials
+): IAMClient {
+  if (!iam) {
+    iam = new IAMClient({
+      region,
+      credentials,
+    });
```

Comment:
shouldn't it be possible to configure this entirely from ENV? so no explicit region or credentials are passed? like we do for the s3 client in utils

Reply:
Confirmed

Reply:
i think viceice wanted to take them from the ENV directly into the get/build client function, and make it a singleton, 
but then i need to INIT the variables for other ways to log in anyway, so it doesn't change complexity.
1) ENV vars
2) config params
3) in the future: get login from home/users/config file or something that was mentioned in the issue